MapboxNavigation/ExitView.swift

```diff
@@ -0,0 +1,120 @@
+import UIKit
+
+enum ExitSide: String{
+    case left, right, other
+    
+    var exitImage: UIImage {
+        return self == .left ? ExitView.leftExitImage : ExitView.rightExitImage
+    }
+}
+
+class ExitView: UIView {
+    static let leftExitImage = UIImage(named: "exit-left", in: .mapboxNavigation, compatibleWith: nil)!.withRenderingMode(.alwaysTemplate)
+    static let rightExitImage = UIImage(named: "exit-right", in: .mapboxNavigation, compatibleWith: nil)!.withRenderingMode(.alwaysTemplate)
+    
+    var side: ExitSide = .right {
+        didSet {
+            populateExitImage()
+            rebuildConstraints()
+        }
+    }
+    
+    lazy var imageView: UIImageView = {
+        let view = UIImageView(image: self.side.exitImage)
+        view.tintColor = .black
+        view.translatesAutoresizingMaskIntoConstraints = false
+        view.contentMode = .scaleAspectFit
+        return view
+    }()
+    
+    lazy var exitNumberLabel: UILabel = {
+        let label: UILabel = .forAutoLayout()
+        label.text = exitText
+        label.textColor = .black
+        print("system font size! \(UIFont.systemFontSize)")
+        label.font = UIFont.boldSystemFont(ofSize: 20)
+        return label
+    }()
+
+    var exitText: String? {
+        didSet {
+            exitNumberLabel.text = exitText
+            invalidateIntrinsicContentSize()
+        }
+    }
+    var pointSize: CGFloat {
+        didSet {
+            rebuildConstraints()
+        }
+    }
+    
+    convenience init(pointSize: CGFloat, side: ExitSide = .right, text: String) {
+        self.init(frame: .zero)
+        self.pointSize = pointSize
+        self.side = side
+        self.exitText = text
+        commonInit()
+    }
+    
+    override init(frame: CGRect) {
+        pointSize = 0.0
+        super.init(frame: frame)
+    }
+    
+    required init?(coder aDecoder: NSCoder) {
+        pointSize = 0.0
+        super.init(coder: aDecoder)
+        commonInit()
+    }
+    
+    func rebuildConstraints() {
+        NSLayoutConstraint.deactivate(self.constraints)
+        buildConstraints()
+    }
+    
+    func commonInit() {
+        backgroundColor = .clear
+        layer.borderWidth = 1.0
+        layer.borderColor = UIColor.black.cgColor
```

Comment:
This should be an appearance property so that we can style in differently in the `NightStyle`.

Reply:
🤦‍♂️ I knew I was forgetting something! Will do.